xarray/core/types.py

```diff
@@ -187,6 +187,9 @@ def copy(
 T_Chunks = Union[int, dict[Any, Any], Literal["auto"], None]
 T_NormalizedChunks = tuple[tuple[int, ...], ...]
 
+DataVars = Mapping[Any, Any]
```

Comment:
Do you think we can figure out a more precise value type?
Maybe DaCompatible or so?

Reply:
That's the hope, but it usually opens several cans of worms when changing away from Any. 

This PR was mainly just to get _MergeResult to pass through the dataset.__init__ so I can continue in the other T_DuckArray PRs.

Reply:
Always the same problem....